lib/metadataTypes/MetadataType.js

```diff
@@ -311,6 +311,59 @@ class MetadataType {
         return;
     }
 
+    /**
+     * test if metadata was actually changed or not to potentially skip it during deployment
+     *
+     * @param {TYPE.MetadataTypeItem} cachedVersion cached version from the server
+     * @param {TYPE.MetadataTypeItem} metadata item to upload
+     * @param {string} [fieldName] optional field name to use for identifying the record in logs
+     * @returns {boolean} true if metadata was changed
+     */
+    static hasChanged(cachedVersion, metadata, fieldName) {
+        // should be set up type by type but the *_generic version is likely a good start for many types
+        return true;
+    }
+    /**
+     * test if metadata was actually changed or not to potentially skip it during deployment
+     *
+     * @param {TYPE.MetadataTypeItem} cachedVersion cached version from the server
+     * @param {TYPE.MetadataTypeItem} metadata item to upload
+     * @param {string} [fieldName] optional field name to use for identifying the record in logs
+     * @param {boolean} [silent] optionally suppress logging
+     * @returns {boolean} true on first identified deviation or false if none are found
+     */
+    static hasChanged_generic(cachedVersion, metadata, fieldName, silent) {
+        if (!cachedVersion) {
+            return true;
+        }
+        // we do need the full set in other places and hence need to work with a clone here
+        const clonedMetada = JSON.parse(JSON.stringify(metadata));
+        this.removeNotUpdateableFields(clonedMetada);
+        // iterate over what we want to upload rather than what we cached to avoid false positives
+        for (const prop in clonedMetada) {
+            if (this.definition.ignoreFieldsForUpdateCheck?.includes(prop)) {
+                continue;
+            }
+            if (Util.isObjectChanged(cachedVersion, metadata, prop)) {
```

Comment:
there is helper libraries which do this. including "assert" which we use for testing. i would recommend we reuse that one, rather than build our own

Reply:
if you can point me towards the right methods? i found it to be difficult given our complex cases

Reply:
For example https://nodejs.org/api/assert.html#assertdeepequalactual-expected-message or https://nodejs.org/api/assert.html#assertequalactual-expected-message could cover this need.